src/firejail/fs_home.c

```diff
@@ -370,6 +370,23 @@ void fs_check_private_dir(void) {
 	}
 }
 
+// check new private working directory (--private-cwd= option) - exit if it fails
+void fs_check_private_cwd(void) {
+	EUID_ASSERT();
+	invalid_filename(cfg.cwd, 0); // no globbing
+
+	// Expand the working directory
+	char *tmp = expand_macros(cfg.cwd);
+	cfg.cwd = realpath(tmp, NULL);
+	free(tmp);
+
+	// is_dir not checked because path may not exist outside of jail
```

Comment:
This comment seems to indicate you want to support paths that don't exist outside the jail, but afaics it won't work because of the realpath(). Just mentioning it in case you need that functionality.

Reply:
good catch - I did not look close enough at the behavior of realpath